### PR TITLE
Update Wayland packages

### PR DESCRIPTION
**Pull request by JL2210**

### CODE REVIEW

File: packages/libwayland.rb [hunk 1]
```diff
@@ -3,33 +3,27 @@
...
+  depends_on 'libxslt'
+  # depends_on 'graphviz' => :build # GraphViz doesn't have PNG support enabled
 
   def self.build
-    system "./configure \
-            --prefix=#{CREW_PREFIX} \
-            --libdir=#{CREW_LIB_PREFIX} \
-            --disable-documentation"
+    system './configure',
+           '--disable-documentation', # See above comment
```
Comment (uberhacker) [inline packages/libwayland.rb]: To a casual observer, they may not understand that `graphviz` is a dependency to build documentation.  Might want to explain that a bit more clearly.

File: packages/wayland_protocols.rb [hunk 2]
```diff
@@ -1,35 +1,20 @@
...
-  })
-  binary_sha256 ({
-    aarch64: 'bd0eb42679be30bddbbe20bb254508be16d8a4f5566c246885fb20a94a2fc3d7',
-     armv7l: 'bd0eb42679be30bddbbe20bb254508be16d8a4f5566c246885fb20a94a2fc3d7',
-       i686: 'ef466cc4ddca3c0d6069c3cdb110d5024b533d78be6512dff377751affe76f94',
-     x86_64: '3d3471f15e2f7e80cef25f272d1f0e29cb0e323e78beaa479df8d319f3b3d477',
-
-  })
-
-  depends_on 'libwayland'
```
[the remark this answers is not in the document]
Comment (JL2210) [reply packages/wayland_protocols.rb]: Because `wayland` already depends on it and it messes up the dependency order.
For example, `wayland_protocols` depends on `wayland-scanner` from `libwayland`. However, if the dependencies are not set up in such a way as they are now, Chromebrew tries to install `wayland_protocols` before `libwayland`.

TL;DR

`libwayland` is already pulled in by `wayland`, so the dependency is redundant.

[the remark this answers is not in the document]
Comment (uberhacker) [reply packages/wayland_protocols.rb]: But isn't wayland a fake package?  What if we just removed it?  I want to get away from fake packages if they aren't necessary.  This looks like that might be the case.

[the remark this answers is not in the document]
Comment (JL2210) [reply packages/wayland_protocols.rb]: Would you be comfortable renaming `wayland_protocols` to `wayland`?

[the remark this answers is not in the document]
Comment (uberhacker) [reply packages/wayland_protocols.rb]: Change the `depends_on` probably makes the most sense.

[the remark this answers is not in the document]
Comment (uberhacker) [reply packages/wayland_protocols.rb]: @JL2210: Were you going to change `depends_on 'wayland'` to `depends_on 'wayland_protocols'` in all the packages?

[the remark this answers is not in the document]
Comment (uberhacker) [reply packages/wayland_protocols.rb]: @JL2210:  You nudge others now it's my turn to nudge you. :)

[the remark this answers is not in the document]
Comment (JL2210) [reply packages/wayland_protocols.rb]: I did. There's only ~3 that depend on `wayland`.

[the remark this answers is not in the document]
Comment (JL2210) [reply packages/wayland_protocols.rb]: Sorry. Must have not pushed it.

**Comment by JL2210** [comment]
Done.

File: packages/wayland.rb [hunk 3]
```diff
@@ -3,10 +3,31 @@
...
+  binary_sha256 ({
+  })
 
-  depends_on 'libwayland'
-  depends_on 'wayland_protocols'
+  depends_on 'expat'
+  depends_on 'libpng'
+  depends_on 'libffi'
+  depends_on 'libxml2'
+  depends_on 'libxslt'
```
Comment (uberhacker) [inline packages/wayland.rb]: Core package so not needed here.

File: packages/wayland.rb [hunk 4]
```diff
@@ -3,10 +3,31 @@
...
+  })
+  binary_sha256 ({
+  })
 
-  depends_on 'libwayland'
-  depends_on 'wayland_protocols'
+  depends_on 'expat'
+  depends_on 'libpng'
+  depends_on 'libffi'
+  depends_on 'libxml2'
```
Comment (uberhacker) [inline packages/wayland.rb]: Core package so not needed here.

Comment (JL2210) [reply packages/wayland.rb]: Got it.
